fix(broker-core): Convert numeric correlation key to string

MiguelPires opened the pull request:
Takes a numeric correlation key from the payload and converts it to a String such that it can be matched with an equivalent String correlation key from a published message.

closes #1667

GitCop commented:
There were the following issues with your Pull Request

 * Commit: 9292073bddf54b0e57074234e79753338427fd79
  * Commits must be in the following format: **%{type}(%{scope}): %{description}**

Guidelines are available at https://github.com/zeebe-io/zeebe/blob/develop/CONTRIBUTING.md

---
This message was auto-generated by https://gitcop.com

saig0 commented on broker-core/src/main/java/io/zeebe/broker/workflow/processor/CatchEventOutput.java:
```diff
@@ -210,7 +213,11 @@ private DirectBuffer extractCorrelationKey(
 
       if (result.isLong()) {
-        return result.getLongAsBuffer();
+        final DirectBuffer buffer = result.getLongAsBuffer();
```
I think we can move this logic into `QueryResults`.

saig0 commented on docs/src/bpmn-workflows/message-events.md:
```diff
@@ -11,7 +11,7 @@ A message can be referenced by one or more message events. It holds the informat
 * the correlation key
 
-The correlation key is specified as JSON Path expression. It is evaluated when the message event is entered and extracts the value from the workflow instance payload. The value must be a string. If the correlation key can't be resolved or it is not a string then an incident is created.
+The correlation key is specified as [JSON Path](reference/json-conditions.html) expression. It is evaluated when the message event is entered and extracts the value from the workflow instance payload. The value must be either a string or a number.
```
I think a sentence is removed by merging ;)

MiguelPires commented:
Moved the conversion logic into the getLongAsString method in QueryResult and deleted getLongAsBuffer since it was no longer used. Also modified the tests and the documentation to reflect the changes.

saig0 commented:
bors r+
